collated/main/A0147971U.md

```diff
@@ -0,0 +1,1319 @@
+     * Modify config file for switchlist command execution.
+     * */
+    public void switchToNewTaskList(String tasklistFileName) throws IOException {
+        addNameToTasklists(tasklistFileName);
+        this.taskListFileName = tasklistFileName + ".xml";
+        String newConfig = JsonUtil.toJsonString(this);
+        PrintWriter newConfigFileWriter = new PrintWriter(DEFAULT_CONFIG_FILE);
+        newConfigFileWriter.write(newConfig);
+        newConfigFileWriter.close();
+    }
+    
+    /**
+     * Replace the old file name with the new one 
+     * in the current file name list. 
+     * @throws TaskListFileExistException 
+     * @throws JsonProcessingException 
+     * @throws FileNotFoundException 
+     * */
+    public void replaceWithNewNameInNameList(String newName) throws IOException, TaskListFileExistException {
+        String[] names = this.getTaskListNames();
+        for (int i=0; i<names.length; i++) {
+            if (names[i].equals(newName)) {
+                throw new TaskListFileExistException();
+            }
+            if ((names[i] + ".xml").equals(this.taskListFileName)) {
+                names[i] = newName;
+            }
+        }
+        this.taskListFileNames = StringUtils.join(names, ", ");
+    }
+    
+    /**
+     * Modify config file for renamelist command execution.
+     * @throws TaskListFileExistException 
+     * @throws IOException 
+     * */
+    public void renameCurrentTaskList(String newTasklistFileName) throws TaskListFileExistException, IOException {
+        replaceWithNewNameInNameList(newTasklistFileName);
+        File newFile = new File(taskListFilePath + "/" + newTasklistFileName + ".xml");
+        File oldFile = new File(taskListFilePath + "/" + this.taskListFileName);
+        oldFile.renameTo(newFile);
+        this.taskListFileName = newTasklistFileName + ".xml";
+        String newConfig = JsonUtil.toJsonString(this);
+        PrintWriter newConfigFileWriter = new PrintWriter(DEFAULT_CONFIG_FILE);
+        newConfigFileWriter.write(newConfig);
+        newConfigFileWriter.close();
+    }
+```
+###### \java\teamfour\tasc\commons\events\storage\FileRelocateEvent.java
+``` java
+package teamfour.tasc.commons.events.storage;
+
+import teamfour.tasc.commons.events.BaseEvent;
+
+public class FileRelocateEvent extends BaseEvent {
+
+    private String destination;
+    
+    public FileRelocateEvent(String destination) {
+        this.destination = destination;
+    }
+    
+    public String getDestination() {
+        return destination;
+    }
+    
+    @Override
+    public String toString() {
+        return this.getClass().getSimpleName();
+    }
+    
+}
+```
+###### \java\teamfour\tasc\commons\events\storage\RequestTaskListRenameEvent.java
+``` java
+package teamfour.tasc.commons.events.storage;
+
+import teamfour.tasc.commons.events.BaseEvent;
+
+/** Indicates the Task List in the model has been renamed*/
+public class RequestTaskListRenameEvent extends BaseEvent {
```

Comment:
if it indicates task list has been renamed; then probably you need to choose the class name to reflect that